File: 17-trees3/4-n-ary-trees/main.py
# N-ary Tree Implementation and Level-Order Traversal - Python Example

class NaryTreeNode:
    def __init__(self, data):
        self.data = data
        self.children = []

def level_order(root):
    if not root:
        return
    from collections import deque
    queue = deque([root])
    while queue:
        node = queue.popleft()
        print(node.data, end=" ")
        for child in node.children:
            queue.append(child)

# Example usage
root = NaryTreeNode(1)
child1 = NaryTreeNode(2)
child2 = NaryTreeNode(3)
child3 = NaryTreeNode(4)

root.children = [child1, child2, child3]
child1.children = [NaryTreeNode(5), NaryTreeNode(6)]

print("Level order traversal of N-ary tree:")
level_order(root)

File: 17-trees3/4-n-ary-trees/main.java
// N-ary Tree Implementation and Level-Order Traversal - Java Example

import java.util.*;

class NaryTreeNode {
    int data;
    List<NaryTreeNode> children;

    NaryTreeNode(int data) {
        this.data = data;
        this.children = new ArrayList<>();
    }
}

class NaryTreeTraversal {
    public static void levelOrder(NaryTreeNode root) {
        if (root == null) return;
        Queue<NaryTreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            NaryTreeNode node = queue.poll();
            System.out.print(node.data + " ");
            for (NaryTreeNode child : node.children) {
                queue.add(child);
            }
        }
    }

    public static void main(String[] args) {
        NaryTreeNode root = new NaryTreeNode(1);
        NaryTreeNode c1 = new NaryTreeNode(2);
        NaryTreeNode c2 = new NaryTreeNode(3);
        NaryTreeNode c3 = new NaryTreeNode(4);

        root.children = Arrays.asList(c1, c2, c3);
        c1.children = Arrays.asList(new NaryTreeNode(5), new NaryTreeNode(6));

        System.out.print("Level order traversal of N-ary tree: ");
        levelOrder(root);
    }
}
